app/models/functional_location.rb

```diff
@@ -0,0 +1,31 @@
+class FunctionalLocation < ApplicationRecord
+
+  validates :uuid, presence: true
+  validates :id_in_source, presence: false
+  validates :description, presence: false
+  validates :short_name, presence: true
+  validates :segment_type, presence: false
+  validates :status_id, presence: false
+  validates :alarm_id, presence: false
+  validates :is_active, presence: false
+  validates :comments, presence: false
+  belongs_to :break_down_structure, optional: true
```

Comment:
There is no need to have fields in both tables pointing to one another. This could have been a `has_one` or `has_many` with the opposite side being the functional location id field of the breakdown structures table.